Bottom: 4786cc9208984e7085d88b9774a0851dbe6f91fc
Top:    069b66e7be6c52093c826731b4c9df1b6e693034
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-16 09:36:17 -0500

Refresh of change-hooks

---
diff --git a/ansible-playbook/group_vars/all/credentials.yml b/ansible-playbook/group_vars/all/credentials.yml
index 69052558..888b3ffb 100644
--- a/ansible-playbook/group_vars/all/credentials.yml
+++ b/ansible-playbook/group_vars/all/credentials.yml
@@ -3,6 +3,7 @@
 buildbot_admin_password:             '{{ lookup("password", playbook_dir + "/credentials/buildbot/admin"                      , length=20) }}'
 buildbot_worker_ntel_password:       '{{ lookup("password", playbook_dir + "/credentials/buildbot/worker_ntel"                , length=20) }}'
 buildbot_worker_ptxdist_password:    '{{ lookup("password", playbook_dir + "/credentials/buildbot/worker_ptxdist"             , length=20) }}'
+buildbot_webhook_secret:             '{{ lookup("password", playbook_dir + "/credentials/buildbot/webhook"                    , length=20) }}'
 buildsystem_database_build_password: '{{ lookup("password", playbook_dir + "/credentials/buildsystem/database/readuser") }}'
 buildsystem_database_root_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/database/root"           , length=20) }}'
 buildsystem_ssh_prod_user_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/ssh/prod"                , length=20) }}'
diff --git a/docker/buildbot-master/buildbot/master.cfg b/docker/buildbot-master/buildbot/master.cfg
index cf6b82fd..c969a528 100644
--- a/docker/buildbot-master/buildbot/master.cfg
+++ b/docker/buildbot-master/buildbot/master.cfg
@@ -92,6 +92,11 @@ c['www'] = {
                 ]),
         ],
     ),
+    'change_hook_dialects': {
+        'gitlab': {
+            'secret': os.getenv('GITLAB_WEBHOOK_SECRET')
+        }
+    }
 }
 
 # PROJECT IDENTITY
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index 89439ecd..ebdabcb4 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -47,6 +47,7 @@ container_env:
   GITLAB_URI: 'https://{{ gitlab_hostname }}/'
   GITLAB_OAUTH_ID: '{{ gitlab_buildbot_auth_id }}'
   GITLAB_OAUTH_SECRET: '{{ gitlab_buildbot_auth_secret }}'
+  GITLAB_WEBHOOK_SECRET: '{{ buildbot_webhook_secret }}'
   LDAP_URI: 'ldap://{{ openldap_hostname }}:389/'
   LDAP_BIND_USER: 'cn=proxyagent,dc=novatech'
   LDAP_BIND_PW: '{{ openldap_proxyagent_password }}'
